catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/WorkspaceTransformer.java

```diff
@@ -88,9 +100,10 @@ private void setupMetacardMappers() {
               List<Map<String, Object>> content = (List) value;
               return content
                   .stream()
+                  .peek(this::removeExternalListAttributes)
```

Comment:
❓  From what I read `peek` is mainly to be used for debugging.

https://stackoverflow.com/questions/33635717/in-java-streams-is-peek-really-only-for-debugging

Reply:
Nice catch; i think filter seems more appropriate if a stream is going to be used

Reply:
✏️ This is a really...awkward way of doing this, in addition to being counter to the intent of the `peek` method. I would suggest 
1. Modify your `removeExternalListAttributes` to take and return a `Map<String, Object>`
2. Change this `peek` to a `map`
3. Since you'll be changing your `removeExternalListAttributes` method, you might consider returning a _new_ map from it or perhaps a filtered view of it. This Guava method might prove helpful for the latter approach - http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/Maps.html#filterKeys-java.util.Map-com.google.common.base.Predicate- - but is certainly not the only way to solve this. Might also be simpler to just perform a stream/filter in that method where your predicate just excludes keys in `EXTERNAL_LIST_ATTRIBUTES`.

Reply:
Fixed.